oemof_b3/tools/testing_pipeline.py

```
@@ -163,83 +181,169 @@ def get_abs_path_list(output_rule_list):
+    Outputs
+    -------
+     None
+
+    """
     # Raw data is needed for some rules and therefore is created if missing
     raw_data_exists = check_raw_data_exists()
 
     # Loop over each rule which is tested in the snakemake pipeline
     for sublist in output_rule_list:
-        # Get absolute path
-        absolute_path_list = [os.path.join(os.getcwd(), entry) for entry in sublist]
+        # Get absolute path of sublist
+        absolute_path_list = get_abs_path_list(sublist)
 
         renamed_path = []
         for raw_file_path in absolute_path_list:
+            try:
+                # Check if file already exists in directory
+                if os.path.isfile(raw_file_path):
+                    # Rename file with extension original
+                    renamed_file = file_name_extension(raw_file_path)
+                    renamed_path.append(renamed_file)
+                else:
+                    # Check for the file with the _original suffix
+                    original_file_path = raw_file_path.replace(
+                        os.path.splitext(raw_file_path)[1],
+                        "_original" + os.path.splitext(raw_file_path)[1],
+                    )
+                    if os.path.exists(original_file_path):
+                        raise FileExistsError(
+                            f"File {original_file_path} already exists."
+                            f"Please rename the file {raw_file_path} first."
+                        )
 
-            if os.path.isfile(raw_file_path):
-                # Get file extension
-                file_extension = raw_file_path[raw_file_path.rfind(".") + 1 :]
-                # Rename existing user data
-                renamed_file = rename_path(
-                    raw_file_path, "." + file_extension, "_original." + file_extension
-                )
-                renamed_path.append(renamed_file)
+            except FileNotFoundError as e:
+                print(e)
+                continue
```

Where does this try and except come from?

---

I integrated a try block to process both dirs and files. The idea was to check if the path exists as file or dir and rename them. If the path does not exist but the path + _original, then it should return an error message. There is also the instance where the file does not exist at all and therefore should continue to the `rule_test`. I reverted the changes and put them into the `pipeline_folder_output_test` .

---

The function has been deleted with the commit [5e3eb48](https://github.com/rl-institut/oemof-B3/pull/357/commits/5e3eb48a6bddc114c941645784807940f40b4105).